projects/adventure/adv.py

```diff
@@ -17,14 +18,216 @@
+      self.length -= 1
+      return item
+  def size(self):
+    return self.length
+
+class Stack():
+    def __init__(self):
+        self.stack = []
+    def push(self, value):
+        self.stack.append(value)
+    def pop(self):
+        if self.size() > 0:
+            return self.stack.pop()
+        else:
+            return None
+    def size(self):
+        return len(self.stack)
+
+visited = set()
+
+def inner_bfs(graph, starting_vertex):
+  global visited
+  q = Queue()
+  q.enqueue([starting_vertex])
+  inner_visited = set()
+  while q.size() > 0:
+    path = q.dequeue()
+    node = path[-1]
+    for destination in graph[node]:
+      if destination not in visited:
+        return path
+    if node not in inner_visited:
+      inner_visited.add(node)
+      for neighbor in graph[node]:
+        new_path = path[:]
+        new_path.append(neighbor)
+        q.enqueue(new_path)
+
+def dft(graph, starting_room):
+  global visited
+  s = Stack()
+  s.push(starting_room.id)
+  main_path = []
+  while s.size() > 0:
+    vertex = s.pop()
+    main_path.append(vertex)
+    if vertex not in visited:
+      visited.add(vertex)
+      if len(visited) == len(graph):
+        return main_path
+      dead_end = True
+      not_visited = []
+      for destination in graph[vertex]:
+        if destination not in visited:
+          dead_end = False
+          not_visited.append(destination)
+      if not dead_end:
+        random_sample = random.sample(not_visited, 1)
+        s.push(random_sample.pop())
+      if dead_end:
+        # The BFS looks for the shortest path from a dead_end vertex to a vertex with at least 1 unexplored neighbor
+        new_path = inner_bfs(graph, vertex)
+        main_path.pop()
+        # Add the path from the BFS to the main path
+        main_path += new_path
+        non_visited = []
+        current_vertex = main_path[-1]
+        for destination in graph[current_vertex]:
+          if destination not in visited:
+            non_visited.append(destination)
+        random_sample = random.sample(non_visited, 1)
+        s.push(random_sample.pop())
+  return main_path
+
+def generatePath(graph, new_player):
+  global visited
+  shortestPath = []
+  for i in range(1000):
+    resultPath = dft(graph, new_player.currentRoom)
+    if i == 0:
+      shortestPath = resultPath
+    elif len(resultPath) < len(shortestPath):
+      shortestPath = resultPath
+    visited.clear()
+  return shortestPath
+
+generatedPath = generatePath(modifiedMap, player)
+# print(generatedPath)
+# print(len(generatedPath))
+
+# Converting the generated path back into cardinal directions
+# This is done by looking at the conversionMap object, and looking for the value at i-1. Inside this object, I look for the current index, and take the cardinal direction that is its value.
+
+for i in range(len(generatedPath)):
+  if i > 0:
+    direction = conversionMap[generatedPath[i-1]][generatedPath[i]]
+    traversalPath.append(direction)
```

Comment:
Good job solving this problem last week. 

I'm curious why you added the DLL for your queue. Python has a built in Stack and Queue that you could use as well.